docs/v3/core/index.rst

```diff
@@ -349,6 +350,25 @@ terminology for a use case of reading from an array:
 .. image:: terminology-read.excalidraw.png
   :width: 600
 
+*Extension points*
+
+    Locations within a `metadata document_` where extension-related
+    metadata can be found. Current extension points are listed in the core spec,
+    e.g. `codecs`, `data_type`. See :ref:`extension points <extensions_section>` below.
+
+*Extensions*
+
+    Components defined in a `metadata document`_ to
+    configure how metadata are interpreted by implementations. These
+    components include codecs, data types, chunk key encodings, chunk grids and
+    storage transformers. See :ref:`extension points <extensions_section>` below.
+
+*Core*
+
+    Core indicates a feature or concepts defined within the Zarr v3
+    specification as defined in this repository. Note, however, that certain
+    core features are explicitly marked as optional for implementations.
```

Comment:
```suggestion
    Core metadata is that which is defined in the Zarr v3 specification.
```

Reply:
I think "core" is used in more than just metadata (other than that basically everything in zarr.json is metadata). The more important part of this is that core != MUST.

Reply:
Why do we need to make a negative statement about what core is not? It should be sufficient to say what it is.

Reply:
Because there is some confusion on this topic. I'll push another formulation.